### ISSUE TITLE
TextEditor: in Dialog submits on key "ENTER"

### ISSUE THREAD

**Issue by Fennes3** [primefaces/primefaces#2832]
## 1) Environment
- PrimeFaces version: 6.1
- The issue is present in 6.0 too

## 2) Expected behavior
Pressing "ENTER" in text-editor causes a line-break and NO form-submit.

## 3) Actual behavior
Form is submitted on "ENTER" in text-editor.

## 4) Steps to reproduce
Place a p:dialog inside a h:form. Insert a p:textEditor and a p:defaultCommand inside the Dialog.
Open Dialog. Press "ENTER" in text-editor -> form gets submitted

## 5) Sample XHTML
```
<h:form ...>
     ...
     <p:dialog id="dialogID" ...>
          ...
          <p:textEditor .../>
          <p:defaultCommand target="buttonID" scope="dialogID" />
          <p:commandLink id="buttonID" type="submit" process="dialogID".../>
    </p:dialog>
    ...
</h:form>
```
## 6) Sample Bean
Unnecessary

Note: Placing the h:form inside the dialog or leave out p:defaultCommand is not an option

**Comment by jxmai** [comment]
This issue is causing by the default behavior of quill editor, and I have yet to find a clean solution since the keyboard event is hidden somewhere.

https://stackoverflow.com/questions/32495936/quill-js-how-to-prevent-newline-entry-on-enter-to-submit-the-input

The answer was from quill maintainer, but I would say that's just a workaround.

Notice that he also mentioned that `A removeHotkey API has been added on the keyboard module but this is not released an any official version yet.`

If you search the API in this page: https://quilljs.com/docs/modules/keyboard/

There is still no such API as of today.

The only two events we currently have https://quilljs.com/0.20/docs/events/#text-change have been implemented, but there is no access to the keyboard event.


---

Pasted some fully running code here for anyone who wants to replicate the bug

```xhtml
<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml"
	xmlns:ui="http://java.sun.com/jsf/facelets"
	xmlns:f="http://java.sun.com/jsf/core"
	xmlns:p="http://primefaces.org/ui"
	xmlns:h="http://java.sun.com/jsf/html">

<h:head>
	<title>PrimeFaces Test</title>
</h:head>
<h:body>

	<h1>#{testView.testString}</h1>


	<p:commandButton value="Open Dialog" oncomplete="PF('test').show()"></p:commandButton>

	<h:form id="testForm">
		<p:dialog id="dialogID" widgetVar="test">
          <p:textEditor/>
          <!-- <p:editor></p:editor> -->
          <p:defaultCommand target="buttonID" scope="dialogID" />
          <p:commandLink id="buttonID" type="submit" process="dialogID"/>
    	</p:dialog>
	</h:form>
</h:body>
</html>
```

**Comment by melloware** [comment]
Another possible issue related on the Quill site for not being able to intercept keyboard events:  https://github.com/quilljs/quill/issues/983

**Comment by jxmai** [comment]
Good reference :+1:  It looks like that `not exposing keyboard events` is by design.

> The event object does contain more information than is passed onto the keyboard binding handlers but does present a leaky abstraction

Some random thoughts: the argument above is hard to debate since the term `leaky abstraction` itself is subjective. When dealing with some legacy codebase for example, some abstraction implementation might be right at its time but after more and more requirements add-on, a new feature might borrow some existed api for various reasons. 

Of course the easiest solution is trying to minimise the number of exposed APIs as possible, but that will reduce the end user's flexibility. 

So If no one can convince quill maintainer by showing him an example representing the benefit of overriding keyboard events externally, and the benefit can be greater than the risk of what's called `leaky abstraction` in the long run, I don't think they are willing to expose keyboard events.

**Comment by melloware** [comment]
Looks like this could help? https://github.com/quilljs/quill/issues/479

**Comment by jxmai** [comment]
@melloware This issue might be fixed. Pls correct me if I am wrong.

I tried to use the sample I provided https://github.com/primefaces/primefaces/issues/2832#issuecomment-343242737 with an additional `secure="false"` due to https://github.com/primefaces/primefaces/issues/5163#issuecomment-532231472

```html
<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml"
	xmlns:ui="http://java.sun.com/jsf/facelets"
	xmlns:f="http://java.sun.com/jsf/core"
	xmlns:p="http://primefaces.org/ui"
	xmlns:h="http://java.sun.com/jsf/html">

<h:head>
	<title>PrimeFaces Test</title>
</h:head>
<h:body>

	<h1>#{testView.testString}</h1>
	
	<p:commandButton value="Open Dialog" oncomplete="PF('test').show()"></p:commandButton>

	<h:form id="testForm">
		<p:dialog id="dialogID" widgetVar="test">
          <p:textEditor secure="false"/>
          <!-- <p:editor></p:editor> -->
          <p:defaultCommand target="buttonID" scope="dialogID" />
          <p:commandLink id="buttonID" type="submit" process="dialogID"/>
    	</p:dialog>
	</h:form>
	
</h:body>
</html>
```

Enter key works inside the dialog with default command

![image](https://user-images.githubusercontent.com/10467831/69442401-29cb8e80-0d12-11ea-9184-f2bec60a9531.png)

---

The solution was introduced in https://github.com/primefaces/primefaces/issues/4984. Related ticket: https://github.com/primefaces/primefaces/issues/5348

**Comment by melloware** [comment]
You are 100% correct!  Marking as fixed in 8.0 and closing ticket.
